pkg/didcomm/protocol/didexchange/keys_test.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package didexchange
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/hyperledger/aries-framework-go/pkg/doc/did"
+	"github.com/hyperledger/aries-framework-go/pkg/kms"
+	mockstorage "github.com/hyperledger/aries-framework-go/pkg/mock/storage"
+)
+
+func TestCreateNewKeyAndVM(t *testing.T) {
+	k := newKMS(t, mockstorage.NewMockStoreProvider())
+
+	t.Run("createNewKeyAndVM success", func(t *testing.T) {
+		didDoc := &did.Doc{}
+
+		err := createNewKeyAndVM(didDoc, kms.ED25519, kms.X25519ECDHKWType, k)
+		require.NoError(t, err)
+		require.Equal(t, ed25519VerificationKey2018, didDoc.VerificationMethod[0].Type)
+		require.Equal(t, x25519KeyAgreementKey2019, didDoc.KeyAgreement[0].VerificationMethod.Type)
+	})
+
+	t.Run("createNewKeyAndVM invalid keyType export signing key", func(t *testing.T) {
+		didDoc := &did.Doc{}
+
+		err := createNewKeyAndVM(didDoc, kms.HMACSHA256Tag256Type, kms.X25519ECDHKWType, k)
+		require.EqualError(t, err, "createAndExportPubKeyBytes: failed to export new public key bytes: "+
+			"exportPubKeyBytes: failed to export marshalled key: exportPubKeyBytes: failed to get public keyset "+
+			"handle: keyset.Handle: keyset.Handle: keyset contains a non-private key")
```

Comment:
Looks like you have some repetition (`keyset.Handle: keyset.Handle:`) in your full error message. If you rework the error messages at each level where they get wrapped you should be able to avoid this

Reply:
anything after `exportPubKeyBytes: failed to get public keyset handle: `  is inside Tink unfortunately